### PR TITLE
Add spring cloud sample service bus messaging

### PR DESCRIPTION
**Pull request by wujack778**
Add spring-cloud-sample-messaging-servicebus module. This module is used to demo how to use @AzureMessageListener annotation with Azure Service Bus to subscribe messages.

### CODE REVIEW

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Spring Cloud Azure Service Bus Messaging Sample shared library for Java
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md:1]: Let's change this module's name to azure-spring-cloud-sample-messaging-servicebus, and for the other one, change it to xxx-messaging-eventhubs.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      <artifactId>azure-spring-cloud-starter-servicebus</artifactId>
+      <version>2.3.0</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.azure.spring</groupId>
+      <artifactId>azure-spring-cloud-messaging</artifactId>
+      <version>2.3.0</version>
+    </dependency>
+```
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md:39]: Do we need to add the version tag here, so the update version script could pick up these versions.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md (hunk 1 above)
Comment (wujack778) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md:1]: all right

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md (hunk 2 above)
Comment (wujack778) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-servicebus-messaging/README.md:39]: Just like this? ` <version>2.3.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-cloud-messaging;current} -->`

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/README.md [hunk 3]
```diff
@@ -28,13 +28,13 @@ dependency `azure-spring-cloud-messaging`.
     <dependency>
       <groupId>com.azure.spring</groupId>
       <artifactId>azure-spring-cloud-starter-servicebus</artifactId>
-      <version>2.3.0</version>
+      <version>2.3.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-cloud-starter-servicebus;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/README.md:31]: I'm not sure whether this is the correct way to add the tag, please refer to this  https://github.com/Azure/azure-sdk-for-java/blob/master/CONTRIBUTING.md#tooling-version-files-and-marker-tags
image

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/README.md [hunk 4]
```diff
@@ -28,13 +28,13 @@ dependency `azure-spring-cloud-messaging`.
...
     </dependency>
 
     <dependency>
       <groupId>com.azure.spring</groupId>
       <artifactId>azure-spring-cloud-messaging</artifactId>
-      <version>2.3.0</version>
+      <version>2.3.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-cloud-messaging;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/README.md:37]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/pom.xml [hunk 5]
```diff
@@ -5,14 +5,14 @@
...
     <artifactId>spring-boot-starter-parent</artifactId>
-    <version>2.4.4</version>
+    <version>2.4.4</version> <!-- {x-version-update;org.springframework.boot:spring-boot-starter-parent;external_dependency} -->
     <relativePath/> <!-- lookup parent from repository -->
   </parent>
   <groupId>com.azure.spring</groupId>
-  <artifactId>azure-spring-cloud-sample-servicebus-messaging</artifactId>
-  <version>1.0.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-cloud-sample-servicebus-messaging;current} -->
+  <artifactId>azure-spring-cloud-sample-messaging-servicebus</artifactId>
+  <version>1.0.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-cloud-sample-messaging-servicebus;current} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/pom.xml:13]: should we also update the version_client.txt?

Comment (saragluna) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/pom.xml:13]: Oh, seems like we don't have an entry there.

Comment (saragluna) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/pom.xml:13]: https://github.com/Azure/azure-sdk-for-java/blob/master/eng/versioning/version_client.txt

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/README.md (hunk 3 above)
Comment (wujack778) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging-servicebus/README.md:31]: get

**Review by saragluna — APPROVED** [review]
LGTM.

**Comment by saragluna** [comment]
The ci kept failing because of azure-messaging-servicebus, which is not caused by this PR, so I am going to override the error and merge this PR.

**Pull request merged**